resources/Microsoft.Windows.Setting.Time/Microsoft.Windows.Setting.Time.psm1

```diff
@@ -0,0 +1,231 @@
+    #     "Saratov Standard Time", "Georgian Standard Time", "Caucasus Standard Time", "Afghanistan Standard Time", "West Asia Standard Time", 
+    #     "Ekaterinburg Standard Time", "Pakistan Standard Time", "India Standard Time", "Sri Lanka Standard Time", "Nepal Standard Time", 
+    #     "Central Asia Standard Time", "Bangladesh Standard Time", "Omsk Standard Time", "Myanmar Standard Time", "SE Asia Standard Time", 
+    #     "Altai Standard Time", "W. Mongolia Standard Time", "North Asia Standard Time", "N. Central Asia Standard Time", 
+    #     "Tomsk Standard Time", "China Standard Time", "North Asia East Standard Time", "Singapore Standard Time", "W. Australia Standard Time", 
+    #     "Taipei Standard Time", "Ulaanbaatar Standard Time", "North Korea Standard Time", "Aus Central W. Standard Time", 
+    #     "Transbaikal Standard Time", "Tokyo Standard Time", "Korea Standard Time", "Yakutsk Standard Time", "Cen. Australia Standard Time", 
+    #     "AUS Central Standard Time", "E. Australia Standard Time", "AUS Eastern Standard Time", "West Pacific Standard Time", 
+    #     "Tasmania Standard Time", "Vladivostok Standard Time", "Lord Howe Standard Time", "Bougainville Standard Time", 
+    #     "Russia Time Zone 10", "Magadan Standard Time", "Norfolk Standard Time", "Sakhalin Standard Time", "Central Pacific Standard Time", 
+    #     "Russia Time Zone 11", "New Zealand Standard Time", "UTC+12", "Fiji Standard Time", "Kamchatka Standard Time", 
+    #     "Chatham Islands Standard Time", "UTC+13", "Tonga Standard Time", "Samoa Standard Time", "Line Islands Standard Time"
+    # )]
+    [string] $TimeZone = (Get-TimeZone).Id
+
+    [DscProperty()]
+    [TimeZoneAutomatically] $SetTimeZoneAutomatically = [TimeZoneAutomatically]::NTP
+
+    [DscProperty()]
+    [nullable[bool]] $ShowSystemTrayDateTime
+
+    static hidden [string] $SetTimeZoneAutomaticallyProperty = 'Type'
+    static hidden [string] $ShowSystemTrayDateTimeProperty = 'ShowSystrayDateTimeValueName'
+
+    [Time] Get()
+    {
+        $currentState = [Time]::New()
+        $currentState.SetTimeZoneAutomatically = [Time]::GetTimeZoneAutoUpdateStatus()
+        $currentState.TimeZone = (Get-TimeZone).Id
+        $currentState.ShowSystemTrayDateTime = [Time]::GetShowSystemTrayDateTimeStatus()
+
+        return $currentState
+    }
+
+    [void] Set()
+    {
+        if ($this.Test())
+        {
+            return
+        }
+
+        $currentState = $this.Get()
+
+        if ($currentState.SetTimeZoneAutomatically -ne $this.SetTimeZoneAutomatically)
+        {
+            Set-ItemProperty -Path $global:tzAutoUpdatePath -Name ([Time]::SetTimeZoneAutomaticallyProperty) -Value $this.SetTimeZoneAutomatically
+        }
+
+        if ($currentState.TimeZone -ne $this.TimeZone)
+        {
+            Set-TimeZone -Id $this.TimeZone    
+        }
+
+        if ($currentState.ShowSystemTrayDateTime -ne $this.ShowSystemTrayDateTime)
```

Comment:
You should be checking if the user actually requested to set these. I invoked the DSC resource without setting this property and the clock in my tray disappeared